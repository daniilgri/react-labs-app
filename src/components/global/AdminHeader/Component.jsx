import React, { useState, useEffect, useRef } from "react";
import { faBars } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import {
  Wrapper,
  BackLink,
  MenuItem,
  Menu,
  StyledNavLink,
  MenuToggle,
} from "./styles";

const Component = () => {
  const [activeMenu, setActiveMenu] = useState(false);
  const [activeProfileSelect, setActiveProfileSelect] = useState(false);

  const profileSelectRef = useRef(null);

  const handleMenuToggleOnClick = (event) => {
    setActiveMenu((prevState) => !prevState);
  };

  const handleClickOutside = (event) => {
    if (
      activeProfileSelect &&
      profileSelectRef.current &&
      !profileSelectRef.current.contains(event.target)
    ) {
      setActiveProfileSelect(false);
    }
  };

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  });

  return (
    <Wrapper role="custom-dropdown">
      <MenuToggle onClick={handleMenuToggleOnClick}>
        <FontAwesomeIcon icon={faBars} />
      </MenuToggle>
      <BackLink to="/profile">Go to profile</BackLink>
      <Menu active={activeMenu}>
        <MenuItem>
          <StyledNavLink to="/admin/users">Users</StyledNavLink>
        </MenuItem>
        <MenuItem>
          <StyledNavLink exact to="/admin/films">
            Films
          </StyledNavLink>
        </MenuItem>
        <MenuItem>
          <StyledNavLink to="/admin/films/new">New film</StyledNavLink>
        </MenuItem>
      </Menu>
    </Wrapper>
  );
};

export default Component;
