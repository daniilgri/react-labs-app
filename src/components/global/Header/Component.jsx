import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { faBars, faUserCircle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import {
  Wrapper,
  LogoWrapper,
  MenuItem,
  Menu,
  StyledNavLink,
  MenuToggle,
  Select,
  SelectIcon,
  OptionsList,
  Option,
  StyledLink,
  OptionLink,
} from "./styles";

const Component = ({ user, loading, error, signOutRequested }) => {
  const [activeMenu, setActiveMenu] = useState(false);
  const [activeProfileSelect, setActiveProfileSelect] = useState(false);

  const profileSelectRef = useRef(null);

  const handleMenuToggleOnClick = (event) => {
    setActiveMenu((prevState) => !prevState);
  };

  const handleProfileSelectOnClick = (event) => {
    setActiveProfileSelect((prevState) => !prevState);
  };

  const handleClickOutside = (event) => {
    if (
      activeProfileSelect &&
      profileSelectRef.current &&
      !profileSelectRef.current.contains(event.target)
    ) {
      setActiveProfileSelect(false);
    }
  };

  const handleDefaultCloseOptionOnClick = (event) => {
    setActiveProfileSelect(false);
  };

  const handleSignOutOptionOnClick = (event) => {
    event.preventDefault();
    signOutRequested();
    setActiveProfileSelect(false);
  };

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  });

  return (
    <Wrapper role="custom-dropdown">
      <MenuToggle onClick={handleMenuToggleOnClick}>
        <FontAwesomeIcon icon={faBars} />
      </MenuToggle>
      <LogoWrapper>Logo</LogoWrapper>
      <Menu active={activeMenu}>
        <MenuItem>
          <StyledNavLink to="/">All films</StyledNavLink>
        </MenuItem>
      </Menu>
      {!loading && (
        <Select active={activeMenu} ref={profileSelectRef}>
          <SelectIcon onClick={handleProfileSelectOnClick}>
            <FontAwesomeIcon icon={faUserCircle} />
          </SelectIcon>
          <OptionsList active={activeProfileSelect}>
            {user ? (
              <React.Fragment>
                <OptionLink
                  to="/profile"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Profile
                </OptionLink>
                <OptionLink
                  to="/profile/orders"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Orders
                </OptionLink>
                <Option onClick={handleSignOutOptionOnClick}>Sign out</Option>
              </React.Fragment>
            ) : (
              <React.Fragment>
                <OptionLink
                  to="/login"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Login
                </OptionLink>
                <OptionLink
                  to="/signup"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Sign up
                </OptionLink>
              </React.Fragment>
            )}
          </OptionsList>
        </Select>
      )}
    </Wrapper>
  );
};

Component.defaultProps = {
  user: null,
  loading: false,
  error: "",
  signOutRequested: () => {},
};

Component.propTypes = {
  user: PropTypes.object,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  signOutRequested: PropTypes.func.isRequired,
};

export default Component;
