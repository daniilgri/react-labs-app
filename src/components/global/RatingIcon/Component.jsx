import React, { useMemo } from "react";
import PropTypes from "prop-types";
import StarIcon from "../StarIcon";
import { Wrapper } from "./styles";

const Component = ({ index, rating, hoverRating, onMouseEnter, onMouseLeave, onSaveRating }) => {
  const fill = useMemo(() => {
    if (hoverRating >= index) {
      return "yellow";
    }
    if (!hoverRating && rating >= index) {
      return "yellow";
    }
    return "none";
  }, [rating, hoverRating, index]);

  const handleMouseEnter = () => {
    onMouseEnter(index);
  };

  const handleMouseLeave = () => {
    onMouseLeave();
  };

  const handleIconOnClick = () => {
    onSaveRating(index);
  };

  return (
    <Wrapper
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      onClick={handleIconOnClick}
    >
      <StarIcon fill={fill} />
    </Wrapper>
  );
};

Component.propTypes = {
  index: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  hoverRating: PropTypes.number.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
  onSaveRating: PropTypes.func.isRequired,
};

export default Component;
