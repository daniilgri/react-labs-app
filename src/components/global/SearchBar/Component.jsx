import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { useFormik } from "formik";

import { Wrapper, InputWrapper, Input, IconWrapper } from "./styles";

const Component = ({ onSet, value }) => {
  const formik = useFormik({
    initialValues: {
      query: value,
    },
    onSubmit: values => {
      onSet(values);
    },
  });

  return (
    <Wrapper onSubmit={formik.handleSubmit}>
      <InputWrapper>
        <Input
          placeholder="Search..."
          name="query"
          id="filmsBoardSearchBar"
          onChange={formik.handleChange}
          value={formik.values.query}
        />
        <IconWrapper type="submit">
          <FontAwesomeIcon icon={faSearch} />
        </IconWrapper>
      </InputWrapper>
    </Wrapper>
  );
};

Component.defaultProps = {
  value: "",
};

Component.propTypes = {
  value: PropTypes.string,
  onSet: PropTypes.func.isRequired,
};

export default Component;
