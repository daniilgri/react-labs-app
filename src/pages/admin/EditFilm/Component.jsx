import React from "react";
import { useParams } from "react-router-dom";

import AdminHeader from "../../../components/global/AdminHeader";
import Form from "../../../components/related/admin/EditFilm/Form";
import { Wrapper } from "./styles";

const Component = () => {
  const { id } = useParams();

  return (
    <>
      <AdminHeader />
      <Wrapper>
        <Form filmId={id} />
      </Wrapper>
    </>
  );
};

export default Component;
