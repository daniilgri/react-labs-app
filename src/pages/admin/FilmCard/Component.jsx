import React from "react";
import { useParams } from "react-router-dom";

import AdminHeader from "../../../components/global/AdminHeader";
import Card from "../../../components/related/admin/FilmCard/Card";
import Subscribers from "../../../components/related/admin/FilmCard/Subscribers";
import { Wrapper } from "./styles";

const Component = () => {
  const { id } = useParams();

  return (
    <>
      <AdminHeader />
      <Wrapper>
        <Card filmId={id} />
        <Subscribers filmId={id} />
      </Wrapper>
    </>
  );
};

export default Component;
