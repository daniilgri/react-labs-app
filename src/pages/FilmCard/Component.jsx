import React from "react";
import { useParams } from "react-router-dom";

import Header from "../../components/global/Header";
import Card from "../../components/related/FilmCard/Card";
import { Wrapper } from "./styles";

const Component = () => {
  const { id } = useParams();

  return (
    <React.Fragment>
      <Header />
      <Wrapper>
        <Card filmId={id} />
      </Wrapper>
    </React.Fragment>
  );
};

export default Component;
