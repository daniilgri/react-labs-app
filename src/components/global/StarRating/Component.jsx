import React, { useState } from "react";
import PropTypes from "prop-types";
import { v4 as uuidv4 } from "uuid";

import RatingIcon from "../RatingIcon";
import { Wrapper } from "./styles";

const Component = ({ count, rate, onRateSet }) => {
  const [hoverRating, setHoverRating] = useState(0);

  const onMouseEnter = index => {
    setHoverRating(index);
  };

  const onMouseLeave = () => {
    setHoverRating(0);
  };

  const onSaveRating = index => {
    onRateSet(index);
  };

  return (
    <Wrapper>
      {Array.from(Array(count).keys()).map(index => {
        return (
          <RatingIcon
            key={uuidv4()}
            index={index + 1}
            rating={rate}
            hoverRating={hoverRating}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
            onSaveRating={onSaveRating}
          />
        );
      })}
    </Wrapper>
  );
};

Component.propTypes = {
  count: PropTypes.number.isRequired,
  rate: PropTypes.number.isRequired,
  onRateSet: PropTypes.func.isRequired,
};

export default Component;
