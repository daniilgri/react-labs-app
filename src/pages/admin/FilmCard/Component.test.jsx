import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import Component from "./Component";

const mockStore = configureStore([]);

describe("admin FilmCard component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      auth: {
        user: {},
      },
      filmCardAdminPanel: {
        loading: false,
        error: "",
      },
      filmSubscribers: {
        count: 1,
        error: "",
        loading: false,
        users: [{}],
      },
    });
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Component />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
