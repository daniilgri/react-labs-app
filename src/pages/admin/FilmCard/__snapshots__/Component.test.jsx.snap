// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin FilmCard component renders correctly 1`] = `
Array [
  <nav
    className="sc-bdfBwQ jLEyRP"
  >
    <div
      className="sc-jSgupP NXinf"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <a
      className="sc-gsTCUz fKNCie"
      href="/profile"
      onClick={[Function]}
    >
      Go to profile
    </a>
    <ul
      className="sc-dlfnbm uDvuY"
    >
      <li
        className="sc-hKgILt kYjJwg"
      >
        <a
          aria-current={null}
          className="sc-eCssSg iyHtsY"
          href="/admin/users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
      <li
        className="sc-hKgILt kYjJwg"
      >
        <a
          aria-current={null}
          className="sc-eCssSg iyHtsY"
          href="/admin/films"
          onClick={[Function]}
        >
          Films
        </a>
      </li>
      <li
        className="sc-hKgILt kYjJwg"
      >
        <a
          aria-current={null}
          className="sc-eCssSg iyHtsY"
          href="/admin/films/new"
          onClick={[Function]}
        >
          New film
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="sc-iUuytg bQiixx"
  >
    <div
      className="sc-lmoMRL EImyq"
    >
      <div
        className="sc-iBPRYJ clCQaR"
      >
        <div
          className="sc-fubCfw fCdnLb"
          src=""
        />
        <div
          className="sc-pFZIQ lcNgUe"
        >
          <ul
            className="sc-crrsfI fSNQeo"
          />
          <div
            className="sc-bqyKva qAJYd"
          >
            <h5
              className="sc-hBEYos kdWGZO"
            >
              Rate 
              <span
                className="sc-kstrdz kekgPR"
              >
                -
              </span>
            </h5>
          </div>
          <h4
            className="sc-iqHYGH imAXwf"
          >
            
             $
          </h4>
          <h2
            className="sc-jrAGrp cfdjqk"
          >
            
          </h2>
          <p
            className="sc-kEjbxe"
          >
            
          </p>
        </div>
      </div>
      <div
        className="sc-fodVxV duiFmH"
      >
        <h3
          className="sc-fFubgz cLfBhW"
        >
          Ticketing
        </h3>
        <div
          className="sc-bkzZxe gAHmgU"
        >
          <div
            className="sc-idOhPF Fpyxh"
          >
            <h5
              className="sc-dIUggk iEZvcG"
            >
              Day
            </h5>
            <ul
              className="sc-hHftDr hDRliZ"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-cBNfnY hVTtNn"
    >
      <div
        className="sc-giIncl jaGXDd"
      >
        <div
          className="sc-ezrdKe fgqbUN"
          src="https://picsum.photos/800/1200"
        />
        <div
          className="sc-bYEvPH eHKRPl"
        >
          <h4
            className="sc-kLgntA kaLjqA"
          >
             
          </h4>
          <p
            className="sc-iktFzd"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
